### ISSUE TITLE
[standard:function-literal] Lambda arrow is removed too aggressively, produces non-compiling code

### ISSUE THREAD

**Issue by mateuszkwiecinski** [pinterest/ktlint#2465]
Probably related to https://github.com/pinterest/ktlint/issues/2331

## Expected Behavior

I expect the code compile after ktlint reformats the file

## Observed Behavior

Formatted file contains code that doesn't compile

## Steps to Reproduce

1. Run format on the following code:
```kotlin
fun repro(foo: Foo) {
    foo.onClick = if (Math.random() > 0.5) { -> } else null
}

class Foo(var onClick: (() -> Unit)?)
```
2. The lambda is removed 
<img width="775" alt="image" src="https://github.com/pinterest/ktlint/assets/36954793/85e5ec57-bedd-4645-b329-d9c7ebcac390">

3. Compilation fails with `Type mismatch: inferred type is Unit but (() -> Unit)? was expected` 

The alternative syntax would be:
```kotlin
foo.onClick = if (Math.random() > 0.5) {
    { }
} else {
    null
}
```
but it doesn't fit in a single line, so it may not be preferred in some cases

## Your Environment

* Version of ktlint used: `1.1.0`
* Relevant parts of the `.editorconfig` settings
```editorconfig
ktlint_code_style = intellij_idea
ktlint_experimental = enabled
```
(I believe this should be enough)
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task):
* Version of Gradle used (if applicable):
* Operating System and version:

**Comment by paul-dingemans** [comment]
Tnx for reporting. the one liner seems to have more problems from Ktlint's perspective, but autocorrect should never resulted in compilation errors.
